Documentation/DesignGuidelines.md

```diff
@@ -6,28 +6,459 @@ Guidelines](http://blogs.msdn.com/b/rxteam/archive/2010/10/28/rx-design-guidelin
+
+#### Events cannot be sent recursively
+
+Just like RAC guarantees that [events will not be received
+concurrently](#events-are-serial), it also guarantees that they won’t be
+received recursively. As a consequence, [operators][] and [observers][] _do not_ need to
+be reentrant.
+
+If an event is sent upon a signal from a thread that is _already processing_
+a previous event from that signal, deadlock will result. This is because
+recursive signals are usually programmer error, and the determinacy of
+a deadlock is preferable to nondeterministic race conditions.
+
+When a recursive signal is explicitly desired, the recursive event should be
+time-shifted, with an operator like [`delay`][delay], to ensure that it isn’t sent from
+an already-running event handler.
+
+#### Events are sent synchronously by default
+
+RAC does not implicitly introduce concurrency or asynchrony. [Operators][] that
+accept a [scheduler][Schedulers] may, but they must be explicitly invoked by the consumer of
+the framework.
+
+A “vanilla” signal or producer will send all of its events synchronously by
+default, meaning that the [observer][Observers] will be synchronously invoked for each event
+as it is sent, and that the underlying work will not resume until the event
+handler finishes.
+
+This is similar to how `NSNotificationCenter` or `UIControl` events are
+distributed.
+
+## The `Signal` contract
+
+A [signal][Signals] is an “always on” stream that obeys [the `Event`
+contract](#the-event-contract).
+
+`Signal` is a reference type, because each signal has identity—in other words, each
+signal has its own lifetime, and may eventually terminate. Once terminated,
+a signal cannot be restarted.
+
+#### Signals start work when instantiated
+
+[`Signal.init`][Signal.init] immediately executes the generator closure that is passed to it.
+This means that side effects may occur even before the initializer returns.
+
+It is also possible to send [events][] before the initializer returns. However,
+since it is impossible for any [observers][] to be attached at this point, any
+events sent this way cannot be received.
+
+#### Observing a signal does not have side effects
+
+The work associated with a `Signal` does not start or stop when [observers][] are
+added or removed, so the [`observe`][observe] method (or the cancellation thereof) never
+has side effects.
+
+A signal’s side effects can only be stopped through [a terminating
+event](#signals-are-retained-until-a-terminating-event-occurs).
+
+#### All observers of a signal see the same events in the same order
+
+Because [observation does not have side
+effects](#observing-a-signal-does-not-have-side-effects), a `Signal` never
+customizes events for different [observers][]. When an event is sent upon a signal,
+it will be [synchronously](#events-are-sent-synchronously-by-default)
+distributed to all observers that are attached at that time, much like
+how `NSNotificationCenter` sends notifications.
```

Comment:
This is mostly a dupe of the event delivery paragraph, but I guess probably worth reiterating.